File: backend/prototipo/src/main/java/com/pedidos/prototipo/ProviderService.java
    package com.pedidos.prototipo;

    import lombok.RequiredArgsConstructor;
    import org.springframework.stereotype.Service;
    import java.util.List;

@Service
@RequiredArgsConstructor
public class ProviderService {

    private final ProviderRepository repo;

    public Provider create(Provider p){ return repo.save(p); }

    public List<Provider> list(){ return repo.findAll(); }

    public Provider get(Long id){
        return repo.findById(id).orElseThrow(NotFoundException::new);
    }

    public Provider update(Long id, Provider data){
        Provider p = get(id);
        p.setName(data.getName());
        p.setContact(data.getContact());
        p.setAddress(data.getAddress());
        return repo.save(p);
    }

    public void delete(Long id){
        repo.deleteById(id);
    }
}

File: frontend/frontend25/src/components/Providers/providerService.js
import axiosClient from "../../api/axiosClient";

const providerService = {
  getAll: () => axiosClient.get("/providers"),
  getById: (id) => axiosClient.get(`/providers/${id}`),
  create: (data) => axiosClient.post("/providers", data),
  update: (id, data) => axiosClient.put(`/providers/${id}`, data),
  delete: (id) => axiosClient.delete(`/providers/${id}`)
};

export default providerService;
